src/routers/TabRouter.js

```diff
@@ -32,6 +32,8 @@ export default (
   const initialRouteIndex = order.indexOf(initialRouteName);
   const backBehavior = config.backBehavior || 'initialRoute';
   const shouldBackNavigateToInitialRoute = backBehavior === 'initialRoute';
+  const shouldBackNavigateToLastActiveTab = backBehavior === 'previousRoute';
+  const backstack = [];
```

Comment:
Okay, makes sense to keep router stateless. 

So just to make sure I understand, is this what you would like to have after selecting a tab:

{ 
 index: 1
 routes: [ route array as is it now ],
 backstack: [ array of routes (key and routename) of tabs previously selected ]
}

So an extra property for it in the state?